internal/bundler/bundler_test.go

```diff
@@ -1,42 +1,102 @@
-/*
-Package bundler provides a transfer package creator for Archivematica.
-
-It is an alternative to TransferSession that addresses some of its major pain
-points. For example, this version does not concern with submitting the package
-to Archivematica and makes the package filesystem available to the user for
-more flexibility to alter its contents manually.
-
-It could eventually be included in the amclient module as its own package.
-*/
-package bundler
+package bundler_test
 
 import (
 	"bytes"
 	"os"
+	"path/filepath"
+	"syscall"
 	"testing"
 
-	"github.com/spf13/afero"
+	"gotest.tools/v3/assert"
+	"gotest.tools/v3/fs"
+
+	"github.com/artefactual-sdps/enduro/internal/bundler"
 )
 
 func TestNewBundlerWithTempDir(t *testing.T) {
-	transferDir, err := os.MkdirTemp("", "")
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	b, err := NewBundlerWithTempDir(transferDir)
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	ok, err := afero.DirExists(b.fs, "")
-	if err != nil {
-		t.Fatal(err)
-	}
-	if !ok {
-		t.Fatal("dir not exists")
-	}
-
-	_ = b.Write("foobar.txt", bytes.NewReader([]byte("foooooo")))
-	_ = b.Bundle()
+	t.Parallel()
+
+	tmpDir := fs.NewDir(t, "enduro-bundler")
+	b, err := bundler.NewBundlerWithTempDir(tmpDir.Path())
+	assert.NilError(t, err)
+
+	// Describe the transfer.
+	b.Describe("dc.title", "Transfer")
+
+	// Describe and write foobar.txt and notes.txt into the transfer.
+	err = b.Write("foobar.txt", bytes.NewReader([]byte("some text")))
+	assert.NilError(t, err)
+	b.DescribeFile("foobar.txt", "dc.title", "Foobar")
+	b.ChecksumMD5("foobar.txt", "552e21cd4cd9918678e3c1a0df491bc3")
+	b.ChecksumSHA1("foobar.txt", "37aa63c77398d954473262e1a0057c1e632eda77")
+	b.ChecksumSHA256("foobar.txt", "b94f6f125c79e3a5ffaa826f584c10d52ada669e6762051b826b55776d05aed2")
+	err = b.Write("notes.txt", bytes.NewReader([]byte("some notes")))
+	assert.NilError(t, err)
+	b.DescribeFile("notes.txt", "dc.title", "Notes")
+	b.DescribeFile("notes.txt", "dc.description", "Some notes")
+
+	err = b.Bundle()
+	assert.NilError(t, err)
+
+	// The actual transfer directory being bundled is created two levels deep,
+	// using a root container ("c") and a temporary directory, for example:
+	// `/tmp/enduro-bundler-2161969315/c/304672139`.
+	dir := transferDir(t, tmpDir.Path())
+	assert.Equal(t, b.FullBaseFsPath(), dir)
+
+	umask := os.FileMode(syscall.Umask(0))
+	dirMode := fs.WithMode(0o755 &^ umask)
```

Comment:
Huh, this is a cool way to check file modes in different environments that have different umasks. :thinking:

Reply:
I think that it needs to be used with caution because it mutates the process umask, but in this case the process is only used for the bundle package, so it should not contaminate other tests. I think that a better alternative would be to update the implementation of the bundler to do a chmod to ensure that permissions are set correctly regardless of the process umask.